src/fsharp/FSharp.Build/Microsoft.FSharp.Targets

```
@@ -474,44 +474,24 @@ this file.
                             Condition="'%(Reference.Identity)' == 'FSharp.Data.TypeProviders' and
-                                       '%(Reference.HintPath)' == '$(_OldRefAssemTPLocation)' and
-                                       Exists('$(NewFSharpCompilerLocation)FSharp.Data.TypeProviders.dll')" />
+                                       $([System.String]::new('%(Reference.HintPath)').EndsWith('$(_OldRefAssemTPLocation)', System.StringComparison.OrdinalIgnoreCase))" />
         <ReferenceToInboxTP Include="@(Reference)"
                             Condition="'%(Reference.Identity)' == 'FSharp.Data.TypeProviders' and
                                        $([System.String]::new('%(Reference.HintPath)').StartsWith('$(_OldSdkTPLocationPrefix)', System.StringComparison.OrdinalIgnoreCase)) and
-                                       $([System.String]::new('%(Reference.HintPath)').EndsWith('$(_OldSdkTPLocationSuffix)', System.StringComparison.OrdinalIgnoreCase)) and
-                                       Exists('$(NewFSharpCompilerLocation)FSharp.Data.TypeProviders.dll')" />
-
-        <!-- Update references to FSharp.Data.TypeProviders.dll -->
-        <Reference Condition="'%(Reference.Identity)' == 'FSharp.Data.TypeProviders' and
-                              '%(Reference.HintPath)' == '$(_OldRefAssemTPLocation)' and
-                              Exists('$(NewFSharpCompilerLocation)FSharp.Data.TypeProviders.dll')">
-          <HintPath>$(NewFSharpCompilerLocation)FSharp.Data.TypeProviders.dll</HintPath>
-        </Reference>
+                                       $([System.String]::new('%(Reference.HintPath)').EndsWith('$(_OldSdkTPLocationSuffix)', System.StringComparison.OrdinalIgnoreCase))" />
 
-        <!-- Update references to FSharp.Data.TypeProviders.dll -->
-        <Reference Condition="'%(Reference.Identity)' == 'FSharp.Data.TypeProviders' and
-                               $([System.String]::new('%(Reference.HintPath)').StartsWith('$(_OldSdkTPLocationPrefix)', System.StringComparison.OrdinalIgnoreCase)) and
-                               $([System.String]::new('%(Reference.HintPath)').EndsWith('$(_OldSdkTPLocationSuffix)', System.StringComparison.OrdinalIgnoreCase)) and
-                               Exists('$(NewFSharpCompilerLocation)FSharp.Data.TypeProviders.dll')">
-
-          <HintPath>$(NewFSharpCompilerLocation)FSharp.Data.TypeProviders.dll</HintPath>
-        </Reference>
       </ItemGroup>
-      <Warning
-          Text="This Project references the obsolete TypeProvider: FSharp.Data.TypeProviders.dll.  Consider Switching to the Nuget package version: https://www.nuget.org/packages/FSharp.Data.TypeProviders."
+      <Error
+          Text="This Project references an obsolete TypeProvider: FSharp.Data.TypeProviders.dll, this was removed In VS2019 16.7.0 Preview 1.  Consider Switching to the Nuget package version: https://www.nuget.org/packages/FSharp.Data.TypeProviders."
```

Use official branding and not be too specific about a preview, as people hit this in 16.8 I think they won't really care about a given preview number.

```suggestion
          Text="This Project references an obsolete TypeProvider: FSharp.Data.TypeProviders.dll, this was removed In Visual Studio 2019 16.7.0. Consider Switching to the NuGet package version: https://www.nuget.org/packages/FSharp.Data.TypeProviders."
```